benchexec/tools/jpf.py

```diff
@@ -34,7 +34,7 @@ def executable(self):
 
 
     def version(self, executable):
-        return 'rev '+self._version_from_tool(executable).split()[2]
+	return open('.version', 'r').read()
```

Comment:
This is highly problematic.

It uses a tab for indentation, which is inconsistent with the rest of the file. Please use spaces.

It does not close the opened file, and thus leaks a file descriptor. Please properly cleanup all resources, preferably using the `with` keyword.

The code looks for `.version` in the current directory instead of the directory of the executable. Thus it will not always find the file. Please change it to look in the correct directory.

If the file cannot be found (for example because the user is benchmarking an older version of JPF), the code will just crash, leaving the user with a stack trace.
Please add proper error handling.

Reply:
I can change the indentation.
Is it really a big problem if the file is not closed? The command is not run very often, and the resource is eventually garbage collected and closed.
Only older versions of JPF would lack the .version file, and older versions are not going to be used, so I do not see a problem here. If we ever use a different mechanism in the future, we can also update jpf.py.
Still, I am of course happy to update the code, if you can point me to example code.
(I cannot run python3 on the only computer I have access to now, and I do not want to make the code overly complex, as then the code might fail in regular situations.)